### PR TITLE
Tickets/dm 35852

### PR DESCRIPTION
**Pull request by teweitsai**
* Add the **.pre-commit-config.yaml**, **pyproject.toml**, and **meta.yaml**.
* Support the **isort**.

### CODE REVIEW

File: conda/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+source:
+  path: ../
+
+build:
+  noarch: generic
+  script: {{ PYTHON }} -m pip install --no-deps --ignore-installed .
+  entry_points:
+    - run_m2gui = lsst.ts.m2gui.application:run_application
+
+requirements:
```
Comment (wvreeven) [inline conda/meta.yaml:15]: I see there is no `test` section in this file. Is that on purpose?

File: conda/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  host:
+    - python {{ python }}
+    - pip
+    - setuptools_scm
+    - setuptools
+    - pytest-runner
```
Comment (wvreeven) [inline conda/meta.yaml:21]: This is no longer required so please delete this line.

File: conda/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    - python {{ python }}
+    - setuptools_scm
+    - setuptools
+    - ts-conda-build =0.3
+  run:
+    - python {{ python }}
+    - setuptools
+    - setuptools_scm
+    - ts-salobj
+    - ts-idl
```
Comment (wvreeven) [inline conda/meta.yaml:32]: Please pin the versions of ts-salobj and ts-idl as follows:

    - ts-idl {{ idl_version }}
    - ts-salobj {{ salobj_version }}

**Review by wvreeven — APPROVED** [review]
Looks good to me. I have left a few comments for your consideration.

File: conda/meta.yaml (hunk 2 above)
Comment (teweitsai) [reply conda/meta.yaml]: Updated in the commit: [6ba959e](https://github.com/lsst-ts/ts_m2gui/pull/18/commits/6ba959ea49c6e6095e71d3670546b4d6c7653eec). Thanks!

File: conda/meta.yaml (hunk 3 above)
Comment (teweitsai) [reply conda/meta.yaml]: Updated in the commit: [6ba959e](https://github.com/lsst-ts/ts_m2gui/pull/18/commits/6ba959ea49c6e6095e71d3670546b4d6c7653eec). Thanks!

**Pull request merged**